airbyte-cdk/python/unit_tests/sources/file_based/scenarios/csv_scenarios.py

```diff
@@ -34,6 +35,146 @@
         }
     )
     .set_file_type("csv")
+    .set_expected_spec(
+        {
+            "documentationUrl": "https://docs.airbyte.com/integrations/sources/in_memory_files",
+            "connectionSpecification": {
+                "title": "InMemorySpec",
+                "description": "Used during spec; allows the developer to configure the cloud provider specific options\nthat are needed "
+                               "when users configure a file-based source.",
+                "type": "object",
+                "properties": {
+                    "streams": {
+                        "title": "The list of streams to sync",
+                        "description": "Streams defines the behavior for grouping files together that will be synced to the downstream "
+                                       "destination. Each stream has it own independent configuration to handle which files to sync, "
+                                       "how files should be parsed, and the validation of records against the schema.",
+                        "order": 10,
+                        "type": "array",
+                        "items": {
+                            "title": "FileBasedStreamConfig",
+                            "type": "object",
+                            "properties": {
+                                "name": {
+                                    "title": "Name",
+                                    "type": "string"
+                                },
+                                "file_type": {
+                                    "title": "File Type",
+                                    "type": "string"
+                                },
+                                "globs": {
+                                    "title": "Globs",
+                                    "type": "array",
+                                    "items": {
+                                        "type": "string"
+                                    }
+                                },
+                                "schemaless": {
+                                    "title": "Schemaless",
+                                    "default": False,
+                                    "type": "boolean"
+                                },
+                                "validation_policy": {
+                                    "title": "Validation Policy",
+                                    "type": "string"
+                                },
+                                "input_schema": {
+                                    "title": "Input Schema",
+                                    "type": "object"
+                                },
+                                "primary_key": {
+                                    "title": "Primary Key",
+                                    "anyOf": [
```

Comment:
@brianjlai have you confirmed the UI can display an `anyOf` like this one? 

FWIW I think this should just be a list of strings. Nested PKs are no longer something we want to encourage, and it will likely be removed from the protocol. List of strings ensures it's just a composite PK